src/GeositeFramework/js/Map.js

```diff
@@ -238,8 +238,12 @@ require(['use!Geosite',
         // body of `updateLegend` immediately looks at visible layers, while some may
         // still asynchronously be in queue to become visible. As a protection, call
         // `updateLegend` explicitly when a layer has finished adding.
-        dojo.connect(esriMap, 'onUpdateEnd', updateLegend);
-        dojo.connect(esriMap, 'onLayerAdd', updateLegend);
+        //
+        // Add legends for tile layers fires both of these events, which results in
+        // the legend flickering (because of our manual redraw), so make sure the
+        // function is not called repetitively.
+        dojo.connect(esriMap, 'onUpdateEnd', _.debounce(updateLegend, 100));
+        dojo.connect(esriMap, 'onLayerAdd', _.debounce(updateLegend, 100));
```

Comment:
Not sure of the intent, but this binds 2 unique debounced functions instead of debouncing all calls to `updateLegend`.